application/common/models/Mfa.php

```diff
@@ -308,12 +312,14 @@ public function verify($value, string $rpOrigin = '', string $verifyType = '', s
      * @param int $userId
      * @param string $type
      * @param string|null $label
+     * @param string $rpOrigin
+     * @param string|null $recoveryEmail
      * @return array
      * @throws BadRequestHttpException
      * @throws ServerErrorHttpException
      * @throws ConflictHttpException
      */
-    public static function create(int $userId, string $type, string $label = null, string $rpOrigin = ''): array
+    public static function create(int $userId, string $type, ?string $label = null, string $rpOrigin = '', ?string $recoveryEmail = ''): array
```

Comment:
Since this is defaulting to an empty string, I suspect it doesn't need to be nullable
```suggestion
    public static function create(int $userId, string $type, ?string $label = null, string $rpOrigin = '', string $recoveryEmail = ''): array
```

Reply:
Unless we change the controller up, it should be nullable. `$recoveryEmail = $req->getBodyParam('recovery_email');` is set to null if it isn't provided.

Reply:
We can easily change that to be an empty string instead of null. I think avoiding null (except when we _really_ need it) will benefit us in the long run.

But, this is almost splitting hairs. Feel free to proceed as seems best to you two on this issue.